pandas/tests/io/json/test_pandas.py

```diff
@@ -756,12 +759,12 @@ def test_series_with_dtype(self):
         "dtype,expected",
         [
             (True, Series(["2000-01-01"], dtype="datetime64[ns]")),
-            (False, Series([946684800000])),
+            (False, Series(["2000-01-01T00:00:00.000"])),
         ],
     )
     def test_series_with_dtype_datetime(self, dtype, expected):
         s = Series(["2000-01-01"], dtype="datetime64[ns]")
-        data = StringIO(s.to_json())
+        data = StringIO(s.to_json(date_format="iso"))
```

Comment:
Instead of changing the arguments you can use `tm.assert_produces_warning`.

Unless we really care about the argument behavior its easier just to keep the default; in the future when the behavior changes a developer knows to come back and remove the `tm.assert_produces_warning` context manager.